microservices/first/utils/docker_runner.py

```diff
@@ -45,35 +45,24 @@ def run_docker_container(
         "docker",
         "run",
         "-d",
-        "--network",
-        "host",
         "-v",
         "/var/run/docker.sock:/var/run/docker.sock",
         "-e",
         f"SPEED_REPO_URL={repo}",
         "-e",
         f"SPEED_REPO_BRANCH={branch}",
         "-e",
-        "SPEED_KAFKA_ADDRESS=localhost:9092",
+        "SPEED_KAFKA_ADDRESS=host.docker.internal:9092",
         "-e",
-        f"SPEED_NUM_WORKERS={str(num_workers)}",
+        f"SPEED_NUM_WORKERS={num_workers}",
         "-e",
-        f"DEPLOYMENT_ID={str(deployment_id)}",
+        f"DEPLOYMENT_ID={deployment_id}",
         img_url,
     ]
 
-    stdout = None
-    # Open subprocess with PIPE to capture output
-    with subprocess.Popen(
-        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-    ) as process:
-        stdout, stderr = process.communicate()
-
-        # Check if the command was successful
-        if process.returncode != 0:
-            raise subprocess.CalledProcessError(
-                process.returncode, command, stderr.decode("utf-8")
-            )
-
-    # Return captured output
-    return stdout.decode("utf-8")[0:12]
+    try:
+        result = subprocess.run(command, check=True, text=True, capture_output=True)
+        return result.stdout.strip()[:12]
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to start Docker container: {e.stderr}")
```

Comment:
I would remove this print since this exception will be logged once it is raised in the Flask app. You could also add an argument for the Flask logger in this function if you'd like